File: day4.lua
counta = 0
countb = 0
for line in io.lines('day4.txt') do
	for n1,m1,n2,m2 in string.gmatch(line,'(%d+)-(%d+),(%d+)-(%d+)') do
		if (n2-n1) * (m2-m1) <= 0 then
			counta = counta + 1
		end
		if (m2-n1) * (n2-m1) < 0 then
			countb = countb + 1
		end
	end
end

print(counta)
print(countb)

File: day4.c
#include<stdio.h>

int main(){
        int n1,m1,n2,m2,counta=0,countb=0;
        FILE* input = fopen("day4.txt","r");
        while(!feof(input)){
                fscanf(input,"%d-%d,%d-%d",&n1,&m1,&n2,&m2);
                counta += (n1-n2)*(m1-m2) <= 0 ? 1 : 0;
                countb += (m2-n1)*(n2-m1) <= 0 ? 1 : 0;
        }
        printf("Part 1:%d\n",counta);
        printf("Part 2:%d\n",countb);
        fclose(input);
}
